yarn-project/p2p/src/testbench/trusted_peers.test.ts

```diff
@@ -0,0 +1,105 @@
+import { createLogger } from '@aztec/foundation/log';
+import { sleep } from '@aztec/foundation/sleep';
+import type { PeerInfo } from '@aztec/stdlib/interfaces/server';
+import { ClientIvcProof } from '@aztec/stdlib/proofs';
```

Comment:
I believe we can probably emulate the desired behaviour with a unit test in the peer manager.
It will run much faster than a longer test here